src/ghci/stderr.rs

```diff
@@ -56,63 +32,8 @@ pub struct GhciStderr {
 }
 
 impl GhciStderr {
-    #[instrument(skip_all, name = "stderr", level = "debug")]
-    pub async fn run(mut self) -> miette::Result<()> {
-        let mut backoff = ExponentialBackoff::default();
-        while let Some(duration) = backoff.next_backoff() {
-            match self.run_inner().await {
-                Ok(()) => {
-                    // MPSC channel closed, probably a graceful shutdown?
-                    tracing::debug!("Channel closed");
-                    break;
-                }
-                Err(err) => {
-                    tracing::error!("{err:?}");
-                }
-            }
-
-            tracing::debug!("Waiting {duration:?} before retrying");
-            tokio::time::sleep(duration).await;
-        }
-
-        Ok(())
-    }
-
-    pub async fn run_inner(&mut self) -> miette::Result<()> {
-        loop {
-            // TODO: Could this cause problems where we get an event and a final stderr line is only
-            // processed after we write the error log?
-            tokio::select! {
-                Ok(Some(line)) = self.reader.next_line() => {
-                    self.ingest_line(line).await;
-                }
-                Some(event) = self.receiver.recv() => {
-                    self.dispatch(event).await?;
-                }
-            }
-        }
```

Comment:
This bit of code waits for lines to be printed on `ghci`'s stderr and ingests them (puts them in a buffer keyed by the current `Mode`). It also waits for events to be received on the input channel. After this refactor, there is no longer any code that calls `ingest_line`, so stderr output is not printed on stderr (similarly, `ghcid.txt` includes no stderr output, just a `Failed, 2521 modules loaded.` line).

Reply:
Alright, I ended up fixing this by keeping the dispatch loop and use of `Sender`s for `stderr.rs`, but left the other stuff mostly the same.